### DISCUSSION TITLE
Prometheus Install Dependencies and Futuring

### DISCUSSION THREAD

**Discussion by BadgerOnABike** [PSaaS-Developers/Project_issues#187] [Team Posts]
Primarily

1) My suggestion is to move away from a "check for version" towards a "try and catch an error" approach by which we no longer pay attention to the version of the oneAPI DPC++/C++ compiler and rather just install and look for any errors that would arise from a failure due to an incorrect version. This would allow users to install any version of oneAPI so long as they exist within the major version of 1.x. Arguably we could do this with other dependencies that are operating on the philosophy of extensible API construction.

Secondarily

2) Move to clang, lose performance, ~gain reliability~ simplify install.

**Comment by RobBryce** [comment]
not gain reliability, simplify install

**Comment by BadgerOnABike** [comment]
I realized that my considered solution is the same for WISE
